File: import-template.py
#!/usr/bin/env python3
"""
Script to import block template content into WordPress database
"""
import subprocess
import sys
import os

# Configuration
template_file = "biederman-wp/templates/front-page.html"
front_page_id = 2  # Update this if your front page has a different ID

# Change to script directory
os.chdir(os.path.dirname(os.path.abspath(__file__)))

try:
    # Read template file
    with open(template_file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Escape single quotes and backslashes for MySQL
    content_escaped = content.replace('\\', '\\\\').replace("'", "''")
    
    # Build MySQL command
    mysql_cmd = f"UPDATE wp_posts SET post_content = '{content_escaped}' WHERE ID = {front_page_id};"
    
    # Execute via docker-compose
    result = subprocess.run(
        ['docker-compose', 'exec', '-T', 'db', 'mysql', '-u', 'wordpress', '-pwordpress', 'wordpress'],
        input=mysql_cmd,
        cwd=os.getcwd(),
        capture_output=True,
        text=True
    )
    
    if result.returncode == 0:
        print(f"✓ Successfully imported template content into page ID: {front_page_id}")
        print(f"✓ You can now edit the page in the WordPress editor at:")
        print(f"  http://localhost:8080/wp-admin/post.php?post={front_page_id}&action=edit")
    else:
        print(f"✗ Error importing template:")
        print(f"  {result.stderr}")
        sys.exit(1)
        
except FileNotFoundError as e:
    print(f"✗ Error: File not found: {e}")
    sys.exit(1)
except Exception as e:
    print(f"✗ Error: {e}")
    sys.exit(1)

File: import-template.php
<?php
/**
 * Script to import block template content into WordPress page
 * 
 * Usage: Run this script via WP-CLI or place it in WordPress root and access via browser
 * 
 * WP-CLI: wp eval-file import-template.php
 * Browser: http://localhost:8080/import-template.php (remove after use!)
 */

// Load WordPress
require_once(__DIR__ . '/biederman-wp/wp-load.php');

// Check if user is admin (for security)
if (!current_user_can('manage_options')) {
    die('You must be logged in as an administrator to run this script.');
}

// Path to template file
$template_file = __DIR__ . '/biederman-wp/templates/front-page.html';

if (!file_exists($template_file)) {
    die("Template file not found: $template_file\n");
}

// Read template content
$template_content = file_get_contents($template_file);

// Get the front page ID
$front_page_id = get_option('page_on_front');

if (!$front_page_id) {
    die("No front page is set. Please set a static front page in Settings > Reading.\n");
}

// Update the page content
$result = wp_update_post(array(
    'ID' => $front_page_id,
    'post_content' => $template_content,
));

if ($result && !is_wp_error($result)) {
    echo "✓ Successfully imported template content into page ID: $front_page_id\n";
    echo "✓ You can now edit the page in the WordPress editor.\n";
    echo "\n";
    echo "Page URL: " . get_edit_post_link($front_page_id) . "\n";
} else {
    echo "✗ Error importing template: " . (is_wp_error($result) ? $result->get_error_message() : 'Unknown error') . "\n";
}

